code/src/components/NewThought.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+
+export const NewThought = ({ newMessage, handleNewThoughtsChange, onFormSubmit }) => {
+  const [remainingChars, setRemainingChars] = useState(160 - newMessage.length);
+
+  const handleInputChange = (event) => {
+    const input = event.target.value;
+    const remaining = 160 - input.length;
```

Comment:
Nice styling to the remaining characters - very sophisticated and smooth with the faded grey color - Good job!

After writing in the input field - the text disappears but the character counter remains. So if I wrote 10 letters - they disappear (without showing the new happy thought) and the counter remains the same.
Only a hard update of the page sets the character counter back to 160, my new happy thought is still not showing up though.

I think it is some small detail in these lines with the handleInputChange that needs something - I'm sorry I can't tell you exactly how to fix it.
I saw another student who put the character-counter in their code like this and the name is different from your code but their "newThought" is the one you call "newMessage" so that souldn't matter.
This student had the fetch for the New Thoughts in this component so it might be an idea to refactor your code this way?
Just an idea - as I mentioned several times I'm not sure exactly whats wrong so I can just give some hints and tips from others code where I can see it works.
I think you are on the right track and that it's just minor details that will solve the bugs in the code.
Here is the example code from another student regarding the character-counter (when the fetch is starting further down I didn't copy-paste the whole fetch since that one is structured like yours but the things above it might be an idea to try out for your New thought-feed? 🥇 

 const HandleFormSubmit = (event) => {
    event.preventDefault();
    if (newThought.length < 5) {
      return alert('Get on that keyboard, more than 5 characters, please!')
    } else if (newThought.length > 140) {
      return alert('Are you some sort of hacker?! No more than 140 characters, please!')
    } else {
    const Submit = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ message: newThought })
      };
      fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts', Submit)